packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -1290,12 +1291,88 @@ flutter:
           expect(link.existsSync(), true);
         }
       }, overrides: <Type, Generator>{
-        FileSystem: () => fs,
+        FileSystem: () => fileSystem,
         ProcessManager: () => FakeProcessManager.any(),
         FeatureFlags: () => featureFlags,
       });
     });
   });
+
+  testWithoutContext('findPlugins sorts the plugin result', () async {
```

Comment:
Do we actually care about this intermediate stage? Adding a test for this means if we want to change it it'll technically be a breaking change, even though this is an internal helper, right?

The thing that I would actually want tests asserting sorting of are files that are checked into people's projects; specifically the issue I know of is the generated registrants. It seems like we should just have test assertions at that level, and not force ordering at intermediate stages.

Reply:
good point (though testing ordering at this level is easier). I did add one case for android, still looking for places where we do assertions on the other platforms.

I don't want to get too deep into refactoring, but it would be easier to read these tests if they were split up by platform where appropriate.